isb_lib/geome_adapter.py

```diff
@@ -96,39 +170,68 @@ def listProjects(self):
     def recordsInProject(self, project_id, record_type):
         L = getLogger()
         L.debug("recordsInProject project %s", project_id)
-        url = f"{GEOME_API}records/{record_type}/json"
+        params = {"includePublic": "true", "admin": "false"}
+        url = f"{GEOME_API}projects/{project_id}/expeditions"
+        #url = f"{GEOME_API}records/{record_type}/json"
         headers = {
             "Accept": "application/json",
         }
         _page_size = 1000
-        params = {
-            "limit": _page_size,
-            "page": 0,
-            "q": f"_projects_:{project_id}",
-        }
-        more_work = True
-        while more_work:
-            response = requests.get(
-                url, params=params, headers=headers, timeout=HTTP_TIMEOUT
+        response = requests.get(
+            url, params=params, headers=headers, timeout=HTTP_TIMEOUT
+        )
+        if response.status_code != 200:
+            L.error(
+                "Unable to load project expeditions project:%s; status: %s; reason: %s",
+                project_id,
+                response.status_code,
+                response.reason,
             )
-            if response.status_code != 200:
-                L.error(
-                    "Unable to load records project:%s; status: %s; reason: %s",
-                    project_id,
-                    response.status_code,
-                    response.reason,
-                )
-                break
-            # L.debug("recordsInProject data: %s", response.text[:256])
-            data = response.json()
-            for record in data.get("content", {}).get(record_type, []):
-                L.debug("recordsInProject Record id: %s", record.get("bcid", None))
-                # print(json.dumps(record, indent=2))
-                # raise NotImplementedError
-                yield record
-            if len(data.get("content", {}).get(record_type, [])) < _page_size:
-                more_work = False
-            params["page"] = params["page"] + 1
+            return
+        # L.debug("recordsInProject data: %s", response.text[:256])
+        expeditions_json = response.json()
+        if len(expeditions_json) == 0:
+            L.error("Project doesn't have any expeditions project:%s", project_id)
+        for expedition_dict in expeditions_json:
+            # Skip developer messages
```

Comment:
In some of the cases, we were getting strings back instead of dicts -- just double check that it's a dictionary before treating it as such.